hibernate-core/src/main/java/org/hibernate/internal/AbstractSharedSessionContract.java

```diff
@@ -602,14 +602,16 @@ public JdbcConnectionAccess getJdbcConnectionAccess() {
 			if ( ! fastSessionServices.requiresMultiTenantConnectionProvider ) {
```

Comment:
The changes in this block are what I was referring to about now-eager generation of the session-identifier